HM4/Car.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  Car.swift
+//  HM4
+//
+//  Created by Dariia Rodari on 11/21/17.
+//  Copyright © 2017 rodariapp. All rights reserved.
+//
+
+import Foundation
+
+enum Disks {
+    case steel
+    case alloy
+    case iron
+}
+
+class Car: NSObject {
+
+    let complectationName: String
+    var disksType: Disks = .iron
+    var isBooked: Bool = false
+    private let leftOnStock: Int
+    private var price: Int
+    var onSale:Bool
+
+    init(complectationName: String,
+         leftOnStock: Int,
+         price: Int,
+         onSale: Bool = false)
+    {
+        self.complectationName = complectationName
+        self.leftOnStock = leftOnStock
+        self.price = price
+        self.onSale = onSale
+    }
+
+
+    func checkIfOnSale() -> Bool {
+        if leftOnStock < 0 || leftOnStock == 1 && isBooked == true {
+            onSale = false
+        } else {
+            onSale = true
+        }
+        return onSale
+    }
+
+
+    func calculatePrice() -> Int {
+        let disksPrice: Int
+        switch disksType {
+        case .alloy:
+            disksPrice = 500
+        case .steel:
+            disksPrice = 1000
+        default:
+            disksPrice = 0
```

Comment:
price for .iron is 0 ?:)
it's better to not use default if you have the only 3rd case which is .iron
you can add another case instead of default